refract-spec.md

```diff
@@ -107,34 +89,511 @@ In addition to expressing Refract Elements as objects with `element`, `attribute
+    ]
+  }
+}
+```
+
+### Prefix (Element)
+
+A prefix is a way to reference elements in prefixed namespaces.
+
+#### Properties
+
+- `element` prefix (string, fixed)
+- `content` (string) - Prefixed name for given namespace
+
+#### Example
+
+This example says the `foo` element is found in the prefixed namespaced prefixed as `ns2`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "prefix": "ns2"
+  },
+  "content": "baz"
+}
+```
+
+## Extending Elements
+
+Elements by default do not include any of the attributes or content from the element in which they reference. An element MAY be extended in order to inherit from other elements.
+
+### Extend (Element)
+
+The `extend` element provides a way to a way to multiply inherit one or more elements to form a new element. The `extend` element MUST NOT affect the original elements being extended, and MUST define a new instance of an element.
+
+The `extend` element MUST do a deep merge of the elements found in the `content`, but MUST NOT include any IDs from the content elements in the final element. Each element found in the `content` MUST derive from the same primitive element type. The elements are merged from first to last.
+
+#### Examples
+
+```json
+{
+  "element": "extend",
+  "content": [
+    {
+      "element": "foo",
+      "attributes": {
+        "baz": "bar"
+      },
+      "content": "first"
+    },
+    {
+      "element": "foo",
+      "content": "second"
+    }
+  ]
+}
+```
+
+The value for this new element would be the following.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "baz": "bar"
+  },
+  "content": "second"
+}
+```
+
+Elements MAY also be referenced when using extend. Here is an element defined and given an ID of `bar`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "id": "bar"
+  },
+  "content": "second"
+}
+```
+
+Here is an extend element referencing the previously `bar` element.
+
+```json
+{
+  "element": "extend",
+  "content": [
+    {
+      "element": "foo",
+      "content": "first"
+    },
+    {
+      "element": "ref",
+      "content": "#bar"
+    }
+  ]
+}
+```
+
+The resulting element would be the following. Note that the `id` was not included in the final element.
+
+```json
+{
+  "element": "foo",
+  "content": "second"
+}
+```
+
+## Creating Element Types
+
+Element types MAY be referenced or extended to create new element types. If given an ID, these elements MAY be used within the local namespace or referenced from other namespaces if accessible from a resolvable URL. Element types defined within a namespace take priority over all other namespaced types.
+
+Once a type is defined and given an ID, the ID MAY be used as an element.
+
+### Type (Element)
+
+The Type element MAY be used to define new elements. It takes an element as its content, and MAY have an ID for the purposes of referencing.
+
+#### Properties
+
+- `element` type (string, fixed)
+- `content` (Element)
+
+#### Example
```

Comment:
I think it my be illuminating to show how the primitives all (at least one or two) inherit from one element contain the attributes properties.